x/ibc/core/02-client/types/msgs_test.go

```diff
@@ -525,6 +693,63 @@ func (suite *TypesTestSuite) TestMarshalMsgSubmitMisbehaviour() {
 	}
 }
 
+func (suite *TypesTestSuite) TestMsgSubmitMisbehaviourGetSignBytes() {
+	var (
+		msg       sdk.Msg
+		err       error
+		expString string
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+	}{
+		{
+			"solo machine client", func() {
+				soloMachine := ibctesting.NewSolomachine(suite.T(), suite.chainA.Codec, "solomachine", "", 2)
+				misbehaviour := soloMachine.CreateMisbehaviour()
+				msg, err = types.NewMsgSubmitMisbehaviour(soloMachine.ClientID, misbehaviour, suite.chainA.SenderAccount.GetAddress())
+				suite.Require().NoError(err)
+				expString = fmt.Sprintf(
+					`{"type":"cosmos-sdk/MsgSubmitMisbehaviour","value":{"client_id":"solomachine","misbehaviour":%s,"signer":"%s"}}`,
+					string(sdk.MustSortJSON(types.ModuleCdc.LegacyAmino.MustMarshalJSON(misbehaviour))),
+					suite.chainA.SenderAccount.GetAddress())
+			},
+		},
+		{
+			"tendermint client", func() {
+				height := types.NewHeight(0, uint64(suite.chainA.CurrentHeader.Height))
+				heightMinus1 := types.NewHeight(0, uint64(suite.chainA.CurrentHeader.Height)-1)
+				header1 := suite.chainA.CreateTMClientHeader(suite.chainA.ChainID, int64(height.RevisionHeight), heightMinus1, suite.chainA.CurrentHeader.Time, suite.chainA.Vals, suite.chainA.Vals, suite.chainA.Signers)
+				header2 := suite.chainA.CreateTMClientHeader(suite.chainA.ChainID, int64(height.RevisionHeight), heightMinus1, suite.chainA.CurrentHeader.Time.Add(time.Minute), suite.chainA.Vals, suite.chainA.Vals, suite.chainA.Signers)
+
+				misbehaviour := ibctmtypes.NewMisbehaviour("tendermint", header1, header2)
+				msg, err = types.NewMsgSubmitMisbehaviour("tendermint", misbehaviour, suite.chainA.SenderAccount.GetAddress())
+				suite.Require().NoError(err)
+				expString = fmt.Sprintf(
+					`{"type":"cosmos-sdk/MsgSubmitMisbehaviour","value":{"client_id":"clientid","client_state":,"signer":"%s"}}`,
+					// string(sdk.MustSortJSON(types.ModuleCdc.LegacyAmino.MustMarshalJSON(misbehaviour))), // FIXME: header panics on amino json marshal
```

Comment:
the misbehaviour headers JSON marshaling is panicking:

```go
panic: Unregistered interface crypto.isPublicKey_Sum [recovered]
	panic: Unregistered interface crypto.isPublicKey_Sum
```

Reply:
@robert-zaremba @clevinson @amaurymartiny @aaronc any hints?

Reply:
you will need to register that interface here https://github.com/cosmos/cosmos-sdk/blob/master/crypto/codec/amino.go. Not sure how other modules do this since it's not registered already.

Reply:
ah I think it's actually the Tendermint pubkey interface:

```
// PublicKey defines the keys available for use with Tendermint Validators
type PublicKey struct {
	// Types that are valid to be assigned to Sum:
	//	*PublicKey_Ed25519
	//	*PublicKey_Secp256K1
	Sum isPublicKey_Sum `protobuf_oneof:"sum"`
}
```

Reply:
oh, I don't fully understand this code, I can take a look in the morning.

The sdk handles tendermint keys as its own, comes in from abci and gets converted before any usage. This should be done here as well to avoid the dependency on tendermint. I remember doing some work a while ago in the sdk to help with this.

Reply:
yeah so the ibc tendermint header imports some types that use the validator proto msgs. Maybe you can coorditate tomorrow morning with @colin-axner?

Reply:
Relaying from this comment from @ebuchman https://github.com/cosmos/gaia/issues/568#issuecomment-767258515, the general consensus seems to be to limit this PR to only include amino support for IBC `MsgTransfer`, and no other IBC messages.

Reply:
I will prune back this pr to just do the change for `MsgTransfer` (I might just open a new pr since that might be easiest). How can we test `MsgTransfer` with a ledger to ensure that everything does actually work fine? Maybe someone can point me to some tools to use?